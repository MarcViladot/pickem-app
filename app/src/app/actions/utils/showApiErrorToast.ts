import {ResponseApiError, WebApiErrorCode} from '../../utils/IResponse';
import {ErrorTools} from '../../utils/ErrorTools';
import Toast from 'react-native-toast-message';
import i18next from 'i18next';

export const SHOW_TOAST = "SHOW_TOAST";
export const showApiErrorToast = (res: ResponseApiError) => {
    const translation = ErrorTools.getApiErrorMsgFromResponse(res);
    Toast.show({
        text1: i18next.t(translation.error, translation.paramsToTranslate),
        type: res.ErrorCode !== -2 ? 'apiError' : 'serverError',
        position: 'bottom',
        props: res
    });
    return {
        type: SHOW_TOAST
    }
};
