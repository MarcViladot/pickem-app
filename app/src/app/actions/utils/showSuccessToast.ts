import i18next from 'i18next';
import Toast from 'react-native-toast-message';

export const SHOW_SUCCESS_TOAST = 'SHOW_SUCCESS_TOAST';
export const showSuccessToast = (msg?: string) => {
  Toast.show({
    text1: msg || i18next.t('COMMON.OPERATION_OK'),
    type: 'success',
    position: 'bottom',
  });
  return {
    type: SHOW_SUCCESS_TOAST,
  };
};
