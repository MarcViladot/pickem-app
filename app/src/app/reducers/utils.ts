// import {SHOW_LOADING} from '../actions/utils/showLoading';
// import {HIDE_LOADING} from '../actions/utils/hideLoading';
// import {SET_LANG} from '../actions/utils/setLang';
import { AnyAction } from "redux";

const initialState = {
    serverLoading: false,
    currentLang: null
};

export default (state = initialState, action: AnyAction) => {
    /*if (action.type === SHOW_LOADING) {
        return {
            ...state,
            serverLoading: true
        };
    } else if (action.type === HIDE_LOADING) {
        return {
            ...state,
            serverLoading: false
        };
    } else if (action.type === SET_LANG) {
        return {
            ...state,
            currentLang: action.lang
        };
    }*/
    return state;
};
