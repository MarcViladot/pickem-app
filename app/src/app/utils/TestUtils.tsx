import store from '../../../store';
import {render} from '@testing-library/react-native'
import {Provider} from 'react-redux';
import React, {useState} from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {DarkTheme, LightTheme, ThemeContext} from '../../../themes';
import {ThemeProvider} from 'styled-components/native';

const AllTheProviders = ({children}) => {
    const [theme, setTheme] = useState('dark');
    const themeData = {theme, setTheme};
    return (
        <Provider store={store}>
            <ThemeContext.Provider value={themeData}>
                <ThemeProvider theme={theme === 'dark' ? DarkTheme : LightTheme}>
                    {children}
                </ThemeProvider>
            </ThemeContext.Provider>
        </Provider>
    )
};

const ReduxProvider = ({children}) => {

    return (
        <Provider store={store}>
            {children}
        </Provider>
    )
};

const ProvidersWithNav = ({children}) => {
    const [theme, setTheme] = useState('dark');
    const themeData = {theme, setTheme};
    return (
        <Provider store={store}>
            <ThemeContext.Provider value={themeData}>
                <ThemeProvider theme={DarkTheme}>
                    <NavigationContainer theme={DarkTheme}>
                        {children}
                    </NavigationContainer>
                </ThemeProvider>
            </ThemeContext.Provider>
        </Provider>
    )
};

const customRender = (ui, options?) =>
    render(ui, {wrapper: AllTheProviders, ...options});

const reduxRender = (ui, options?) =>
    render(ui, {wrapper: ReduxProvider, ...options});

const navRender = (ui, options?) =>
    render(ui, {wrapper: ProvidersWithNav, ...options});

export * from '@testing-library/react-native'
export {customRender as render}
export {reduxRender as reduxRender}
export {navRender as navRender}
