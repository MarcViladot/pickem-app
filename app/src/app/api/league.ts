import { User, UserCredentials } from "../interfaces/user.interface";
import Api from "./api";
import { AxiosError, AxiosResponse } from "axios";
import { ResponseApi, ResponseServerError } from "../utils/IResponse";
import {Round} from '../interfaces/league.interface';

export default {

  getGroupLeague(groupId: number, leagueId: number) {
    return Api.get(`league/getGroupLeague/${groupId}/${leagueId}`)
      .then((r: AxiosResponse<ResponseApi<any>>) => {
        return r.data
      }, (err: AxiosError) => {
        return new ResponseServerError(err);
      })
  },

  getRoundDetail(roundId: number) {
    return Api.get(`round/${roundId}`)
      .then((r: AxiosResponse<ResponseApi<Round>>) => {
        return r.data
      }, (err: AxiosError) => {
        return new ResponseServerError(err);
      })
  }
}
