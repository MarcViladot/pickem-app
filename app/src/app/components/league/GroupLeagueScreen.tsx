import React, {FC, useEffect} from "react";
import {StackNavigationProp} from "@react-navigation/stack";
import {RouteProp, useTheme} from "@react-navigation/native";
import {DrawerStackParamList} from "../DrawerNavigator";
import TabHomeScreen from "./tabs/TabHomeScreen";
import TabRoundsScreen from "./tabs/Rounds/TabRoundsScreen";
import {FontAwesomeIcon} from "@fortawesome/react-native-fontawesome";
import {faBars, faTrophy} from "@fortawesome/free-solid-svg-icons";
import {createBottomTabNavigator} from "@react-navigation/bottom-tabs";
import {StyleSheet} from "react-native";
import {faHome} from "@fortawesome/free-solid-svg-icons/faHome";
import {useTranslation} from "react-i18next";
import {faClock} from "@fortawesome/free-regular-svg-icons";
import TabTableScreen from "./tabs/Table/TabTableScreen";
import TabMoreScreen from "./tabs/More/TabMoreScreen";
import {LeagueInfo} from "../../interfaces/league.interface";

type ScreenNavigationProps = StackNavigationProp<DrawerStackParamList, "GroupLeague">;
type ScreenRouteProp = RouteProp<DrawerStackParamList, "GroupLeague">;

interface Props {
    navigation: ScreenNavigationProps;
    route: ScreenRouteProp;
}

export type TabsStackParamList = {
    TabHome: undefined;
    TabTable: undefined;
    TabMore: undefined;
    TabRounds: undefined;
};
const Tab = createBottomTabNavigator<TabsStackParamList>();

const GroupLeagueScreen: FC<Props> = ({navigation, route}) => {

    const {colors} = useTheme();
    const {t} = useTranslation();
    const {leagueInfo} = route.params;

    return (
        <Tab.Navigator initialRouteName={"TabHome"} screenOptions={{
            headerShown: false,
            tabBarStyle: [styles.tabBarStyle, {backgroundColor: colors.card}],
            tabBarLabelStyle: styles.tabBarLabelStyle,
            tabBarIconStyle: styles.tabBarIconStyle,
        }}>
            <Tab.Screen name="TabHome" component={TabHomeScreen} options={{
                tabBarIcon: ({focused, color}) => <FontAwesomeIcon icon={faHome} color={colors.text} size={15}/>,
                tabBarLabel: t("LEAGUE.HOME")
            }}/>
            <Tab.Screen name="TabRounds"
                        children={props => <TabRoundsScreen {...props} leagueInfo={leagueInfo}/>}
                        options={{
                            tabBarIcon: ({focused, color}) => <FontAwesomeIcon icon={faClock} color={colors.text}
                                                                               size={15}/>,
                            tabBarLabel: t("LEAGUE.ROUNDS")
                        }}/>
            <Tab.Screen name="TabTable"
                        children={props => <TabTableScreen {...props} leagueInfo={leagueInfo}/>}
                        options={{
                            tabBarIcon: ({focused, color}) => <FontAwesomeIcon icon={faTrophy} color={colors.text}
                                                                               size={15}/>,
                            tabBarLabel: t("LEAGUE.TABLE")
                        }}/>
            <Tab.Screen name="TabMore"
                        children={props => <TabMoreScreen {...props} leagueInfo={leagueInfo}/>} options={{
                tabBarIcon: ({focused, color}) => <FontAwesomeIcon icon={faBars} color={colors.text} size={15}/>,
                tabBarLabel: t("LEAGUE.MORE")
            }}/>
        </Tab.Navigator>
    );
};

const styles = StyleSheet.create({
    tabBarStyle: {
        position: "absolute",
        top: 0,
        // Remove border top on both android & ios
        borderTopWidth: 0,
        elevation: 0,
        shadowColor: "#5bc4ff",
        shadowOpacity: 0,
        shadowOffset: {
            width: 0, height: 0 // for iOS
        },
        shadowRadius: 0,
        height: 40
    },
    tabBarLabelStyle: {
        marginVertical: 4,
        fontSize: 12
    },
    tabBarIconStyle: {
        marginTop: 5
    }
});

export default GroupLeagueScreen;
