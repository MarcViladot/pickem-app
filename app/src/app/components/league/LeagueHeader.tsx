import React, {FC, useEffect} from "react";
import {StyleSheet, Text, TouchableOpacity, View} from "react-native";
import {useSelector} from "react-redux";
import {RootState} from "../../reducers";
import UserImage from "../common/UserImage";
import {FontAwesomeIcon} from "@fortawesome/react-native-fontawesome";
import {DrawerHeaderProps} from "@react-navigation/drawer/lib/typescript/src/types";
import {faBell, faDotCircle} from "@fortawesome/free-regular-svg-icons";
import {faBell as faBellSolid, faChevronLeft} from "@fortawesome/free-solid-svg-icons";
import {ILeagueInfo} from "../../interfaces/league.interface";
import styled from 'styled-components/native';
import {StackHeaderProps} from '@react-navigation/stack';
import {useTranslation} from 'react-i18next';
import {useTheme} from '@react-navigation/native';
import {ThemeText} from '../common/ThemeText';
import { ThemeView } from "../common/ThemeView";

const LeagueHeader: FC<DrawerHeaderProps> = ({scene}) => {

    const { colors } = useTheme();
    const user = useSelector((state: RootState) => state.user.currentUser);

    const {leagueInfo} = scene.route.params as { leagueInfo: ILeagueInfo };

    return (
        <ThemeView style={styles.container}>
            <View style={{flexDirection: "row", alignItems: "center"}}>
                <TouchableOpacity activeOpacity={.5} onPress={() => scene.descriptor.navigation.toggleDrawer()}>
                    <UserImage user={user} styles={styles.userImage}/>
                </TouchableOpacity>
                <ThemeText style={styles.leagueName}>{leagueInfo.groupInfo.name} - {leagueInfo.leagueInfo.name}</ThemeText>
            </View>
            <View>
                <View style={styles.chip}>
                    <FontAwesomeIcon icon={faDotCircle} color={"#ffa600"} size={13}/>
                    <ThemeText style={styles.chipText}>{leagueInfo.totalPoints}</ThemeText>
                </View>
            </View>
        </ThemeView>
    );
};

export const HomeHeader: FC<DrawerHeaderProps> = ({scene}) => {

    const {colors} = useTheme();
    const user = useSelector((state: RootState) => state.user.currentUser);

    const BadgeContainer = styled.View`
      position: absolute;
      right: -3px;
      top: -5px;
      background-color: #38b174;
      border-radius: 8px;
      padding: 0 4px;
      height: 12px;
      justify-content: center;
      align-items: center;
    `;

    return (
        <ThemeView style={styles.container}>
            <View style={{flexDirection: "row", alignItems: "center"}}>
                <TouchableOpacity activeOpacity={.5} onPress={() => scene.descriptor.navigation.toggleDrawer()}>
                    <UserImage user={user} styles={styles.userImage}/>
                </TouchableOpacity>
                <Text style={{marginLeft: 10, fontSize: 20}}>Picke'ms</Text>
            </View>
            {/*<View>
                <TouchableOpacity activeOpacity={.6}
                                  onPress={() => scene.descriptor.navigation.navigate('Notifications')}>
                    <FontAwesomeIcon icon={user.invitations.length ? faBellSolid : faBell} color={colors.text} size={20}/>
                    {!!user.invitations.length && (
                        <BadgeContainer>
                            <Text style={{color: '#fff', fontSize: 10}}>{user.invitations.length}</Text>
                        </BadgeContainer>
                    )}
                </TouchableOpacity>
            </View>*/}
        </ThemeView>
    )
}

export const HomeTitleHeader: FC<DrawerHeaderProps> = ({scene}) => {

    const {colors} = useTheme();
    const {t} = useTranslation();

    return (
        <View style={[styles.container, {height: 45, backgroundColor: colors.card}]}>
            <TouchableOpacity activeOpacity={.5} onPress={() => scene.descriptor.navigation.goBack()}
                              style={styles.backButton}>
                <FontAwesomeIcon icon={faChevronLeft} color={colors.text} size={13}/>
            </TouchableOpacity>
            <ThemeText style={styles.title}>{t(`MENU.${scene.route.name}`)}</ThemeText>
            <View style={{width: 30}}/>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        paddingVertical: 5,
        paddingHorizontal: 10,
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center"
    },
    userImage: {
        width: 35,
        height: 35,
        borderRadius: 50
    },
    chip: {
        flexDirection: "row",
        alignItems: "center",
        borderWidth: .5,
        borderColor: "#e0e0e0",
        borderRadius: 50,
        paddingVertical: 4,
        paddingHorizontal: 10
    },
    chipText: {
        marginLeft: 10,
        fontSize: 13,
        fontWeight: "bold"
    },
    leagueName: {
        fontSize: 17,
        marginLeft: 12
    },
    title: {
        fontSize: 18,
        fontWeight: "bold",
    },
    backButton: {
        borderWidth: 1,
        width: 30,
        height: 30,
        justifyContent: "center",
        alignItems: "center",
        borderRadius: 50,
        borderColor: "#C7C8CA"
    }
});

export default LeagueHeader;
