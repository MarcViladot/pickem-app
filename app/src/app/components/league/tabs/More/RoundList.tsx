import React, {FC} from 'react';
import {Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
import {faChevronLeft, faChevronRight} from '@fortawesome/free-solid-svg-icons';
import {ThemeText} from '../../../common/ThemeText';
import {CommonUtils} from '../../../../utils/CommonUtils';
import {Match, Round} from '../../../../interfaces/league.interface';
import {ThemeView} from '../../../common/ThemeView';
import {format} from 'date-fns';
import styled from 'styled-components/native';
import {useTranslation} from 'react-i18next';
import {useTheme} from '@react-navigation/native';

interface Props {
    rounds: Round[];
    defaultRoundIndex: number;
}

const RoundList: FC<Props> = ({rounds, defaultRoundIndex}) => {

    const {colors} = useTheme();
    const {i18n} = useTranslation();
    const [showRoundIndex, setShowRoundIndex] = React.useState(defaultRoundIndex === -1 ? 0 : defaultRoundIndex);

    const MatchComponent: FC<{ match: Match }> = ({match}) => (
        <ThemeView style={styles.match}>
            <Image source={{uri: match.teams[0].team.crest}} style={styles.teamCrest}/>
            {match.finished ? (
                <ThemeText style={styles.matchResult}
                           testID={"finalResult"}>{match.teams[0].finalResult} Â· {match.teams[1].finalResult}</ThemeText>
            ) : (
                <ThemeText style={styles.matchDate}
                           testID={"startDate"}>{format(new Date(match.startDate), 'dd/MM HH:mm')}</ThemeText>
            )}
            <Image source={{uri: match.teams[1].team.crest}} style={styles.teamCrest}/>
        </ThemeView>
    );

    const IconButton = styled.TouchableOpacity<{ disabled: boolean }>`
      opacity: ${props => props.disabled ? '0.5' : '1'};
      border: 1px solid #C7C8CA;
      width: 30px;
      height: 30px;
      justify-content: center;
      align-items: center;
      border-radius: 50px;
    `;

    const incrementRoundIndex = () => {
        setShowRoundIndex(prevState => {
            if (prevState < rounds.length - 1) {
                return prevState + 1;
            }
            return prevState;
        });
    };

    const decrementRoundIndex = () => {
        setShowRoundIndex(prevState => {
            if (prevState > 0) {
                return prevState - 1;
            }
            return prevState;
        });
    };


    return (
        <>
            {rounds.map((round, i) => (
                showRoundIndex === i && (
                    <View key={i}>
                        <View style={styles.roundHeader}>
                            <IconButton activeOpacity={.5} disabled={i === 0} onPress={() => decrementRoundIndex()} testID={'previousButton'}>
                                <FontAwesomeIcon icon={faChevronLeft} color={colors.text} size={16}/>
                            </IconButton>
                            <ThemeText
                                style={{fontSize: 17}}>{CommonUtils.getRoundName(round, i18n.options.lng)}</ThemeText>
                            <IconButton activeOpacity={.5} disabled={i === rounds.length - 1}
                                        onPress={() => incrementRoundIndex()} testID={'nextButton'}>
                                <FontAwesomeIcon icon={faChevronRight} color={colors.text} size={16}/>
                            </IconButton>
                        </View>
                        <View style={styles.matchesContainer}>
                            {round.matches.map((match, j) => <MatchComponent match={match} key={j}/>)}
                        </View>
                    </View>
                )
            ))}
        </>
    );
};

const styles = StyleSheet.create({
    roundHeader: {
        justifyContent: "space-between",
        flexDirection: "row",
        alignItems: "center"
    },
    matchesContainer: {
        paddingTop: 20,
        flexDirection: "row",
        flexWrap: "wrap",
        justifyContent: "space-between"
    },
    match: {
        width: "49%",
        marginBottom: 10,
        height: 45,
        borderRadius: 50,
        paddingVertical: 5,
        paddingHorizontal: 15,
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center"
    },
    teamCrest: {
        height: 20,
        width: 20,
        resizeMode: "stretch"
    },
    matchResult: {
        fontSize: 12
    },
    matchDate: {
        fontSize: 10
    },
});

export default RoundList;
