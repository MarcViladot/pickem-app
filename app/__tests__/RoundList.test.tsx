import {fireEvent, waitFor} from '@testing-library/react-native';
import {render} from '../src/app/utils/TestUtils';
import React from 'react';
import {Round} from '../src/app/interfaces/league.interface';
import RoundList from '../src/app/components/league/tabs/More/RoundList';

describe('Round List component', () => {

    const rounds: Round[] = [
        {
            id: 1,
            name: 'Round 1',
            startingDate: '2021-12-22T12:19:16.506Z',
            finished: false,
            matches: [],
            visible: true,
            translationGroup: {
                id: 1,
                groupName: "Round",
                roundNames: [
                    {
                        id: 1,
                        text: "Round",
                        lang: "en",
                    },
                    {
                        id: 2,
                        text: "Jornada",
                        lang: "es",
                    }
                ]
            },
            translationNameExtra: '1'
        },
        {
            id: 2,
            name: 'Round 2',
            startingDate: '2021-12-22T12:19:16.506Z',
            finished: false,
            matches: [],
            visible: true,
            translationGroup: {
                id: 1,
                groupName: "Round",
                roundNames: [
                    {
                        id: 1,
                        text: "Round",
                        lang: "en",
                    },
                    {
                        id: 2,
                        text: "Jornada",
                        lang: "es",
                    }
                ]
            },
            translationNameExtra: '2'
        },

    ]

    it('renders correctly', async () => {
        const {toJSON} = await waitFor(() =>
            render(<RoundList rounds={rounds} defaultRoundIndex={0}/>)
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('match should display startDate if not finished', async () => {
        const newMockRound = JSON.parse(JSON.stringify(rounds[0]));
        newMockRound.matches = [
            {
                id: 1,
                startDate: "2021-08-13T19:00:36.000Z",
                finished: false,
                doublePoints: false,
                cancelled: false,
                roundId: 1,
                updatedAt: "2021-12-03T10:56:37.000Z",
                predictions: [
                    {
                        id: 2,
                        localTeamResult: 2,
                        awayTeamResult: 3,
                        correct: false,
                        points: 0,
                        userId: 6,
                        matchId: 1
                    }
                ],
                teams: [
                    {
                        id: 1,
                        finalResult: 1,
                        teamPosition: 0,
                        teamId: 10,
                        team: {
                            id: 10,
                            name: "Valencia",
                            crest: "https://www.flashscore.es/res/image/data/K6UJTt7k-fguToQZ6.png"
                        }
                    },
                    {
                        id: 2,
                        finalResult: 0,
                        teamPosition: 1,
                        teamId: 20,
                        team: {
                            id: 20,
                            name: "Getafe",
                            crest: "https://www.flashscore.es/res/image/data/hGMWMdT0-fguToQZ6.png"
                        }
                    }
                ]
            }
        ];
        const {queryByTestId} = await waitFor(() =>
            render(<RoundList rounds={[newMockRound]} defaultRoundIndex={0}/>)
        );
        const startDate = await queryByTestId('startDate');
        expect(startDate).toBeTruthy();
    });

    it('match should display final result if finished', async () => {
        const newMockRound = JSON.parse(JSON.stringify(rounds[0]));
        newMockRound.matches = [
            {
                id: 1,
                startDate: "2021-08-13T19:00:36.000Z",
                finished: true,
                doublePoints: false,
                cancelled: false,
                roundId: 1,
                updatedAt: "2021-12-03T10:56:37.000Z",
                predictions: [
                    {
                        id: 2,
                        localTeamResult: 2,
                        awayTeamResult: 3,
                        correct: false,
                        points: 0,
                        userId: 6,
                        matchId: 1
                    }
                ],
                teams: [
                    {
                        id: 1,
                        finalResult: 1,
                        teamPosition: 0,
                        teamId: 10,
                        team: {
                            id: 10,
                            name: "Valencia",
                            crest: "https://www.flashscore.es/res/image/data/K6UJTt7k-fguToQZ6.png"
                        }
                    },
                    {
                        id: 2,
                        finalResult: 0,
                        teamPosition: 1,
                        teamId: 20,
                        team: {
                            id: 20,
                            name: "Getafe",
                            crest: "https://www.flashscore.es/res/image/data/hGMWMdT0-fguToQZ6.png"
                        }
                    }
                ]
            }
        ];
        const {queryByTestId} = await waitFor(() =>
            render(<RoundList rounds={[newMockRound]} defaultRoundIndex={0}/>)
        );
        const result = await queryByTestId('finalResult');
        expect(result).toBeTruthy();
    });

    it('should render the default round', async () => {
        const {queryByText} = await waitFor(() =>
            render(<RoundList rounds={rounds} defaultRoundIndex={1}/>)
        );
        const result = await queryByText('Round 2');
        expect(result).toBeTruthy();
    });

    it('should render the second round on next click', async () => {
        const {queryByText, queryByTestId} = await waitFor(() =>
            render(<RoundList rounds={rounds} defaultRoundIndex={0}/>)
        );
        const round2 = await queryByText('Round 2');
        expect(round2).toBeFalsy();
        const next = await queryByTestId('nextButton');
        fireEvent.press(next);
        const round2After = await queryByText('Round 2');
        expect(round2After).toBeTruthy();
    });

    it('should render the first round on previous click', async () => {
        const {queryByText, queryByTestId} = await waitFor(() =>
            render(<RoundList rounds={rounds} defaultRoundIndex={1}/>)
        );
        const round1 = await queryByText('Round 1');
        expect(round1).toBeFalsy();
        const previous = await queryByTestId('previousButton');
        fireEvent.press(previous);
        const round1After = await queryByText('Round 1');
        expect(round1After).toBeTruthy();
    });

    it('should not change round on next button click in the last round', async () => {
        const {queryByText, queryByTestId} = await waitFor(() =>
            render(<RoundList rounds={rounds} defaultRoundIndex={1}/>)
        );
        const round2 = await queryByText('Round 2');
        expect(round2).toBeTruthy();
        const next = await queryByTestId('nextButton');
        fireEvent.press(next);
        const round2After = await queryByText('Round 2');
        expect(round2After).toBeTruthy();
    });

    it('should not change round on previous button click in the first round', async () => {
        const {queryByText, queryByTestId} = await waitFor(() =>
            render(<RoundList rounds={rounds} defaultRoundIndex={0}/>)
        );
        const round1 = await queryByText('Round 1');
        expect(round1).toBeTruthy();
        const previous = await queryByTestId('previousButton');
        fireEvent.press(previous);
        const round1After = await queryByText('Round 1');
        expect(round1After).toBeTruthy();
    });
});
