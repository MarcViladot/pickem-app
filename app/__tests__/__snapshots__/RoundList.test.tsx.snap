// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Round List component renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#C7C8CA",
          "borderRadius": 50,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 30,
          "justifyContent": "center",
          "opacity": 0.5,
          "width": 30,
        }
      }
      testID="previousButton"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={16}
        bbWidth={16}
        fill="#FFF"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        secondaryFill={null}
        secondaryOpacity={0.4}
        size={16}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {},
            Object {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={512}
        vbWidth={320}
        width={16}
      >
        <RNSVGGroup
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
            fill={4294967295}
            fillOpacity={1}
            propList={
              Array [
                "fill",
                "fillOpacity",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
          },
          Object {
            "fontSize": 17,
          },
        ]
      }
    >
      Round 1
    </Text>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#C7C8CA",
          "borderRadius": 50,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "width": 30,
        }
      }
      testID="nextButton"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={16}
        bbWidth={16}
        fill="#FFF"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        secondaryFill={null}
        secondaryOpacity={0.4}
        size={16}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {},
            Object {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={512}
        vbWidth={320}
        width={16}
      >
        <RNSVGGroup
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill={4294967295}
            fillOpacity={1}
            propList={
              Array [
                "fill",
                "fillOpacity",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "paddingTop": 20,
      }
    }
  />
</View>
`;
